__tests__/diet.test.js

```diff
@@ -0,0 +1,34 @@
+import Diet from "../pages/diet.js";
+import {render, screen, fireEvent, act} from "@testing-library/react";
+
+const fetchDiet = require("../apiServices/fetchDiet");
+jest.mock("../apiServices/fetchDiet", () => jest.fn());
```

Comment:
Please refactor the `fetchDiet` function to return the actual data instead of passing setters into it.
Then mock the function to return your fake data.